Sources/Classes/RSMoEngageDestination.swift

```diff
@@ -6,62 +6,89 @@
 //
 
 import Foundation
+
 import Rudder
-import MoEngage
+import MoEngageSDK
+
+
 
 class RSMoEngageDestination: NSObject, RSDestinationPlugin, UNUserNotificationCenterDelegate {
     let type = PluginType.destination
     let key = "MoEngage"
     var client: RSClient?
     var controller = RSController()
+    var moEngageDataCenter: MoEngageDataCenter?
 
-    func update(serverConfig: RSServerConfig, type: UpdateType) {
+    func update(serverConfig: RSServerConfig, type: UpdateType)  {
         guard type == .initial else { return }
         guard let moEngageConfig: RSMoEngageConfig = serverConfig.getConfig(forPlugin: self) else {
             client?.log(message: "Failed to Initialize MoEngage Factory", logLevel: .warning)
             return
         }
+        
+        // Redirect data according to region
+        if moEngageConfig.region == "EU" {
+            moEngageDataCenter = .data_center_02
+        } else if moEngageConfig.region == "US" {
+            moEngageDataCenter = .data_center_01
+        } else if moEngageConfig.region == "IND" {
+            moEngageDataCenter = .data_center_03
+        }
+        else{
+            client?.log(message:"MoEngage SDK initialization terminated due to an invalid region.", logLevel: .warning)
+        }
```

Comment:
As we are terminating, please add ` return` statement after the log statement. Otherwise, it'll continue to initialise the SDK.